server/accounts.go

```diff
@@ -508,33 +517,94 @@ func (a *Account) TrackServiceExportWithSampling(service, results string, sampli
 		return ErrNoSysAccount
 	}
 
-	a.mu.Lock()
-	defer a.mu.Unlock()
-
 	if a == nil {
 		return ErrMissingAccount
 	}
+
+	a.mu.Lock()
 	if a.exports.services == nil {
+		a.mu.Unlock()
 		return ErrMissingService
 	}
 	ea, ok := a.exports.services[service]
 	if !ok {
+		a.mu.Unlock()
 		return ErrMissingService
 	}
 	if ea == nil {
-		ea = &exportAuth{}
+		ea = &serviceExport{}
 		a.exports.services[service] = ea
 	} else if ea.respType != Singleton {
+		a.mu.Unlock()
 		return ErrBadServiceType
 	}
 	ea.latency = &serviceLatency{
 		sampling: int8(sampling),
 		subject:  results,
 	}
+	s := a.srv
+	a.mu.Unlock()
+
+	if s == nil {
+		return nil
+	}
+
+	// Now track down the imports and add in latency as needed to enable.
+	s.accounts.Range(func(k, v interface{}) bool {
+		acc := v.(*Account)
+		acc.mu.Lock()
+		for _, im := range acc.imports.services {
+			if im != nil && im.acc.Name == a.Name && subjectIsSubsetMatch(im.to, service) {
+				im.latency = ea.latency
+			}
+		}
+		acc.mu.Unlock()
+		return true
+	})
 
 	return nil
 }
 
+// UnTrackServiceExport will disable latency tracking of the named service.
+func (a *Account) UnTrackServiceExport(service string) {
+	if a.srv != nil && !a.srv.eventsEnabled() {
+		return
+	}
+
+	a.mu.Lock()
+	if a == nil || a.exports.services == nil {
+		a.mu.Unlock()
+		return
+	}
+	ea, ok := a.exports.services[service]
+	if !ok || ea == nil || ea.latency == nil {
+		a.mu.Unlock()
+		return
+	}
+	// We have latency here.
+	ea.latency = nil
+	a.mu.Unlock()
+
+	if a.srv == nil {
```

Comment:
you usually capture this in the locked section above..

Reply:
Yes will make consistent.